pq-crypto/EVP_kem.h

```diff
@@ -0,0 +1,176 @@
+/********************************************************************************************
+* KEM Library
+*
+* Abstract: Post-Quantum Algorithm API
+*********************************************************************************************/
+
+#ifndef AWSLC_EVP_KEM_H
+#define AWSLC_EVP_KEM_H
+
+#include <stdint.h>
+#include "EVP_pq_constants.h"
+
+/*************************************************
+* Name: pq_kem
+*
+* Description: Keeps track of pq algorithm specific
+* constants which are length of public key, private key,
+* cipher text, and shared secret. Also contains pointers
+* to the specific algorithms required.
+*
+*
+**************************************************/
+
+/* typedef to hide the internal details */
+
+typedef struct pq_kem {
+    // name of pq algorithm specific KEM
+    const char *name;
+    // stores the pq algorithm specific public key memory size
+    const uint16_t public_key_length;
+    // stores the pq algorithm specific private key memory size
+    const uint16_t private_key_length;
+    // stores the pq algorithm specific ciphertext memory size
+    const uint16_t ciphertext_length;
+    // stores the pq algorithm specific shared secret memory size
+    const uint16_t shared_secret_key_length;
+
+
+    /* NIST Post Quantum KEM submissions require the following API for compatibility */
+    int (*generate_keypair)(OUT unsigned char *public_key, OUT unsigned char *private_key);
+    int (*encapsulate)(OUT unsigned char *ciphertext, OUT unsigned char *shared_secret, IN const unsigned char *public_key);
+    int (*decapsulate)(OUT unsigned char *shared_secret, IN const unsigned char *cipher_text, IN const unsigned char *private_key);
+} pq_kem;
+
+/*************************************************
+* Name: pq_kem_params
+*
+* Description: Keeps track of the actual keys and
+* secrets that are pq algorithm specific.
+* They are public key, private key,
+* cipher text, and shared secret. Also points
+* to the algorithm specific pq_kem struct
+*
+*
+**************************************************/
+
+/* typedef to hide the internal details */
+
+typedef struct pq_kem_params {
+    pq_kem *kem;
+    unsigned char *public_key;
+    uint32_t public_key_size;
```

Comment:
I don't think these `size` fields provide anything useful to us. They are always going to be equal to the `length` definitions (as set in `pq_kem_params_alloc`, so this seems like unnecessary duplication of code. 

I understand wanting to adopt s2n's blob functionality to some extent, but short of fully importing the entire blob functionality (which we should not do), I don't think there's an easy way.